app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/PropertyDetailsController.scala

```diff
@@ -726,6 +730,11 @@ class PropertyDetailsController @Inject() (
         _ => routes.PropertyDetailsController.checkYourAnswers()
       )
 
+  private def getCompletionDate(answers: MultipleDisposalsTriageAnswers): CompletionDate =
+    answers
+      .fold(_.completionDate, c => Some(c.completionDate))
+      .getOrElse(CompletionDate(LocalDateUtils.today()))
```

Comment:
we shouldn't default to today - we should redirect to the task list if we can't find a completion date in the triage answers

Reply:
in your pattern match for the tax year you could possible add in a check for the completion date as well at the same time:
```scala
 m.triageAnswers.fold(i => i.taxYear -> i.completionDate, c => Some(c.taxYear) -> Some(c.completionDate))) match {
  case (Some(taxYear), Some(completionDate)) => ...
   case _ => Redirect(taskList)
```

Reply:
Yes I liked this approach